### PR TITLE
Artifact management shortcuts + game begin default keybind fix

### PR DESCRIPTION
**Pull request by dydzio0614**
Added alt+click artifact slot shortcut to move artifact from/to backpack
Added ctrl+click artifact slot shortcut to move artifact to 2nd hero we are trading with

Removed "enter" key as non-campaign game begin shortcut and "b" as campaign begin (to match H3, also begin for standard maps interferes with pregame chat)

### CODE REVIEW

**Comment by dydzio0614** [comment]
@SoundSSGood since this is related to your code, you might want to review

File: client/widgets/CWindowWithArtifacts.cpp [hunk 1]
```diff
@@ -147,8 +147,66 @@ void CWindowWithArtifacts::clickPressedArtPlaceHero(CArtifactsOfHeroBase & artsI
...
+										LOCPLINT->cb->swapArtifacts(ArtifactLocation(artSetPtr->getHero()->id, artSetPtr->getHero()->getSlotByInstance(art)),
+										ArtifactLocation(anotherHeroEquipmentPointer->getHero()->id, availablePosition));
+									}
+								}
+							}
+							else if(GH.isKeyboardAltDown())
+							{
+								ArtifactPosition destinationPosition = ArtifactPosition::PRE_FIRST;
+
+								if(ArtifactUtils::isSlotEquipment(artSetPtr->getHero()->getSlotByInstance(art)))
```
Comment (SoundSSGood) [inline client/widgets/CWindowWithArtifacts.cpp:182]:
```suggestion
								if(ArtifactUtils::isSlotEquipment(artPlace.slot))
```

File: client/widgets/CWindowWithArtifacts.cpp [hunk 2]
```diff
@@ -147,8 +147,66 @@ void CWindowWithArtifacts::clickPressedArtPlaceHero(CArtifactsOfHeroBase & artsI
...
+								{
+									ArtifactPosition availablePosition = ArtifactUtils::getArtBackpackPosition(artSetPtr->getHero(), art->getTypeId());
+									if(availablePosition != ArtifactPosition::PRE_FIRST)
+									{
+										destinationPosition = availablePosition;
+									}
+								}
+								else if(ArtifactUtils::isSlotBackpack(artSetPtr->getHero()->getSlotByInstance(art)))
```
Comment (SoundSSGood) [inline client/widgets/CWindowWithArtifacts.cpp:190]:
```suggestion
								else if(ArtifactUtils::isSlotBackpack(artPlace.slot))
```

File: client/widgets/CWindowWithArtifacts.cpp [hunk 3]
```diff
@@ -147,8 +147,66 @@ void CWindowWithArtifacts::clickPressedArtPlaceHero(CArtifactsOfHeroBase & artsI
...
+
+								if(destinationPosition != ArtifactPosition::PRE_FIRST)
+								{
+									LOCPLINT->cb->swapArtifacts(ArtifactLocation(artSetPtr->getHero()->id, artSetPtr->getHero()->getSlotByInstance(art)),
+										ArtifactLocation(artSetPtr->getHero()->id, destinationPosition));
+								}
+							}
+							else
+							{
+								LOCPLINT->cb->swapArtifacts(ArtifactLocation(artSetPtr->getHero()->id, artSetPtr->getHero()->getSlotByInstance(art)),
```
Comment (SoundSSGood) [inline client/widgets/CWindowWithArtifacts.cpp:207]:
```suggestion
								LOCPLINT->cb->swapArtifacts(ArtifactLocation(artPlace.slot),
```

File: client/widgets/CWindowWithArtifacts.cpp [hunk 4]
```diff
@@ -147,8 +147,66 @@ void CWindowWithArtifacts::clickPressedArtPlaceHero(CArtifactsOfHeroBase & artsI
...
+									ArtifactPosition availablePosition = ArtifactUtils::getArtAnyPosition(artSetPtr->getHero(), art->getTypeId());
+									if(availablePosition != ArtifactPosition::PRE_FIRST && availablePosition != ArtifactPosition::BACKPACK_START)
+									{
+										destinationPosition = availablePosition;
+									}
+								}
+
+								if(destinationPosition != ArtifactPosition::PRE_FIRST)
+								{
+									LOCPLINT->cb->swapArtifacts(ArtifactLocation(artSetPtr->getHero()->id, artSetPtr->getHero()->getSlotByInstance(art)),
```
Comment (SoundSSGood) [inline client/widgets/CWindowWithArtifacts.cpp:201]:
```suggestion
									LOCPLINT->cb->swapArtifacts(ArtifactLocation(artSetPtr->getHero()->id, artPlace.slot),
```

File: client/widgets/CWindowWithArtifacts.cpp [hunk 5]
```diff
@@ -147,8 +147,66 @@ void CWindowWithArtifacts::clickPressedArtPlaceHero(CArtifactsOfHeroBase & artsI
...
+
+								if(destinationPosition != ArtifactPosition::PRE_FIRST)
+								{
+									LOCPLINT->cb->swapArtifacts(ArtifactLocation(artSetPtr->getHero()->id, artPlace.slot),
+										ArtifactLocation(artSetPtr->getHero()->id, destinationPosition));
+								}
+							}
+							else
+							{
+								LOCPLINT->cb->swapArtifacts(ArtifactLocation(artPlace.slot),
```
Comment (SoundSSGood) [inline client/widgets/CWindowWithArtifacts.cpp:207]:
```suggestion
								LOCPLINT->cb->swapArtifacts(ArtifactLocation(artSetPtr->getHero()->id, artPlace.slot),
```
oh sorry, it should be like this

**Pull request merged**